pkg/datapath/linux/routing/info.go

```diff
@@ -56,7 +56,7 @@ func parse(gateway string, cidrs []string, macAddr string) (*RoutingInfo, error)
 		return nil, fmt.Errorf("invalid ip: %s", gateway)
 	}
 
-	if cidrs == nil || len(cidrs) == 0 {
```

Comment:
Why was this nil check removed? It'll make this test fail: https://github.com/cilium/cilium/blob/84db4f63a7c99c472823b84c202196897272fb4a/pkg/datapath/linux/routing/info_test.go#L79

Reply:
@christarazi thanks for your input. I enabled my IDE for _gosimple_ by default, and basically len() for nil slices is defined as zero in go, so I don't think the above test is failing. I just verify by running the test locally as well.

Feel free to let me know otherwise.

Reply:
Ah, I didn't know that!  :)